velox/exec/tests/RowContainerTest.cpp

```diff
@@ -466,29 +465,62 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     }
   }
 
-  template <typename T, typename V>
-  void testOrderAndNullsFirstVariations(
+  template <bool canHandleNulls>
+  void testRowContainerEqualAPI(const TypePtr& type, const VectorPtr& values) {
+    auto numNulls = 0;
+
+    if constexpr (!canHandleNulls) {
+      numNulls = BaseVector::countNulls(values->nulls(), 0, values->size());
+    }
+
+    auto numRows = values->size();
+    auto valuesSlice = values->slice(numNulls, numRows - numNulls);
+    auto rowContainer = makeRowContainer({type}, {type}, false);
+    DecodedVector decoded(*valuesSlice);
+    auto rows = storeRows(decoded, numRows - numNulls, *rowContainer);
+
+    int32_t index{0};
+    for (auto row : rows) {
+      ASSERT_TRUE(rowContainer->equals<canHandleNulls>(
```

Comment:
We should add some tests for rowContainer->equals returning false as well.

Reply:
What kind of values do you think of using to test the evaluation to false? 
This test uses the edge values for floating points so compare them against each other? Or some random floating point values?

Reply:
Yeah, I was thinking one edge against another like say NaN with max/min and also will some regular random floating point values.

Reply:
Ok, I added some more tests to test random and edge values against NaN values.